app/models/bill.rb

```diff
@@ -0,0 +1,41 @@
+class Bill < ActiveRecord::Base
+  has_many :sponsorships
+  has_many :cosponsors, -> { merge(Sponsorship.cosponsored) }, through: :sponsorships, source: :legislator
+  has_one :authorship, -> { merge(Sponsorship.sponsored) }, class_name: 'Sponsorship'
+  has_one :sponsor, through: :authorship, source: :legislator
```

Comment:
it was the only way I could make `has_one :sponsor` work. Can't do it with `through: :sponsorships` because `:sponsorships` is a collection. I'm happy to change it if you know a better way (I don't like `:authorship`, either)